### ISSUE TITLE
[opencv] compile error with opencv-contrib

### ISSUE THREAD

**Issue by itsuhane** [Microsoft/vcpkg#3024]
I tried install opencv with `.\vcpkg install opencv[*]`, and it gives following error during compilation:

```
f:\vcpkg\buildtrees\opencv\x64-windows-dbg\modules\cvv\opencv_cvv_autogen\3jysceobda\../../../../../src/opencv_contrib-3.4.0/modules/cvv/src/view/dual_filter_view.hpp(85): error C2079: 'cvv::view::DualFilterView::rawImages_' uses undefined class 'std::array<cv::Mat,2>'
f:\vcpkg\buildtrees\opencv\x64-windows-dbg\modules\cvv\opencv_cvv_autogen\3jysceobda\../../../../../src/opencv_contrib-3.4.0/modules/cvv/src/view/dual_filter_view.hpp(79): error C2027: use of undefined type 'std::array<cv::Mat,2>'
f:\vcpkg\buildtrees\opencv\x64-windows-dbg\modules\cvv\opencv_cvv_autogen\3jysceobda\../../../../../src/opencv_contrib-3.4.0/modules/cvv/src/view/dual_filter_view.hpp(55): note: see declaration of 'std::array<cv::Mat,2>'
f:\vcpkg\buildtrees\opencv\x64-windows-dbg\modules\cvv\opencv_cvv_autogen\3jysceobda\../../../../../src/opencv_contrib-3.4.0/modules/cvv/src/view/dual_filter_view.hpp(81): error C2664: 'cvv::view::DualFilterView::DualFilterView(const cvv::view::DualFilterView &)': cannot convert argument 1 from 'std::array<cv::Mat,2>' to 'std::array<cv::Mat,2>'
f:\vcpkg\buildtrees\opencv\x64-windows-dbg\modules\cvv\opencv_cvv_autogen\3jysceobda\../../../../../src/opencv_contrib-3.4.0/modules/cvv/src/view/dual_filter_view.hpp(81): note: use of undefined type 'std::array<cv::Mat,2>'
f:\vcpkg\buildtrees\opencv\x64-windows-dbg\modules\cvv\opencv_cvv_autogen\3jysceobda\../../../../../src/opencv_contrib-3.4.0/modules/cvv/src/view/dual_filter_view.hpp(55): note: see declaration of 'std::array<cv::Mat,2>'
```

* I'm using VS2015 Community with Update 3 on Windows 10.
* I'm trying to install opencv 3.4.0-3.
* I'm using `x64-windows` triplet (set in `VCPKG_DEFAULT_TRIPLET`).
* I've installed all dependencies beforehand without trouble.
* I didn't test `x86-windows` and I don't have VS2017.

```
> .\vcpkg version
Vcpkg package management program version 0.0.105-2018-03-12-148cb369d8934d32e37ef8e15c95476ec8e5968a
```

```
> cl
Microsoft (R) C/C++ Optimizing Compiler Version 19.00.24215.1 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.
```

Seems it lacks std::array so I added the missing include (https://github.com/itsuhane/vcpkg/commit/4571234c05cd3fac1dbffbd797c13a8e08a4d8a7). After that x64-windows-dbg compiled, but x64-windows-rel failed. I believe the failure in x64-windows-rel build relates to opencv[cuda] not opencv[contrib].

**Comment by jasjuang** [comment]
This error is actually solved in 3.4.1 as shown in https://github.com/opencv/opencv/pull/10589

As soon as #2906 is merged this problem will go away

**Comment by itsuhane** [comment]
That’s nice. Later I found the build problem with opencv[cuda] was solved in 3.4.1 too.
